snappy_pipeline/workflows/somatic_gene_fusion_calling/__init__.py

```diff
@@ -405,19 +538,18 @@ def get_result_files(self):
         We will process all NGS libraries of all test samples in all sample
         sheets.
         """
-        name_pattern = "{fusion_caller}.{ngs_library.name}"
         for fusion_caller in self.config["tools"]:
+            sub_step = self.sub_steps[fusion_caller]
+            name_patterns = sub_step.get_result_files()
```

Comment:
I suggest keeping the `get_result_files()` isolated from the steps. Easier to figure out what `rule all` needs if it is in just one place. Also, shouldn't they have the same pattern? I don't get why some stay in the `work` directory and others go to `output`...?

Reply:
The different methods produce very different output files. It is difficult/impossible to identify a common format for the output. 

This is the reason why I wanted to give the possibility to modulate the output for each tool. So `LinkOut` just moves the results computed in `work` to `output`.

Reply:
That makes a lot of sense, my comment was about the location. I'd keep it in `SomaticGeneFusionCallingWorkflow` instead of depending on the steps. Something like:

```python
# Convert sheet parsing into method
library_names_list = self._get_all_rna_ngs_libraries()
# Get results
name_pattern = "{fusion_caller}.{ngs_library.name}"
for fusion_caller in self.config["tools"]:  
  for ngs_library in library_names_list:
    # Constant to all callers
    name_pattern_value = name_pattern.format(fusion_caller=fusion_caller, ngs_library=ngs_library)
    yield os.path.join("output", name_pattern_value, "out", ".done")
    # Caller specific stuff...
    if fusion_caller == 'specific_caller':
      yield from self._yield_specific_caller_files()

```

`VariantCallingWorkflow.get_result_files()` is an example